app/controllers/users_controller.rb

```diff
@@ -2,5 +2,11 @@ class UsersController < ApplicationController
 
   def show
     @user = User.find(params[:id])
+    if session[:user_id] == @user.id
+      render 'show'
+    else
+      session[:user_id] = nil
+      redirect_to login_path
+    end
```

Comment:
I like this too. The more comfortable I get with sessions, the more I realize that this is the way to go.